File: github_repo_cloner.rb
require 'net/https'
require 'json'
require 'git'
puts "Please enter your Github account name"
STDOUT.flush
account_name = gets.chomp
response = Net::HTTP.get(URI.parse("https://api.github.com/users/#{account_name}/repos"))
repositories = JSON.parse(response)

repositories.each do |repo|
	git = Git.clone(repo["clone_url"], repo["name"], :path => "my_repositories")
	git.add_remote("originate", repo["clone_url"])
	puts repo["clone_url"]
end

File: old/python/github_repo_cloner.py
import os
import git
import requests

from git import Repo
username = raw_input("Please enter your Github account name: ")
response = requests.get("https://api.github.com/users/" + username + "/repos")
repositories =  response.json()

for repo in repositories:
	command = git.Git().clone(repo["clone_url"], "my_repositories/" + repo["name"], recursive=True)
	r = Repo("my_repositories/" + repo["name"])
	r.create_remote('originate', repo["clone_url"])
	print command